src/Core/Domain/Cart/QueryHandler/GetLastEmptyCustomerCartHandlerInterface.php

```diff
@@ -0,0 +1,43 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\Cart\QueryHandler;
+
+use PrestaShop\PrestaShop\Core\Domain\Cart\Query\GetLastEmptyCustomerCart;
+use PrestaShop\PrestaShop\Core\Domain\Cart\ValueObject\CartId;
+
+/**
+ * Interface for service that handles getting last empty customer cart
+ */
+interface GetLastEmptyCustomerCartHandlerInterface
+{
+    /**
+     * @param GetLastEmptyCustomerCart $query
+     *
+     * @return CartId|null
```

Comment:
@eternoendless in this case cart can be not found, im wondering if returning `null` is ok or better throw `CartNotFoundException`?

Reply:
I think return null is OK.

To me, the system should throw an Exception when something goes wrong. Like a cURL request got no response. Or we have an Order with no Customer ID. But if we **attempts** to get something *in case it exists*, to be returned null is fine.

If you have a `$cartId` and you call `getCartById()` you can expect an exception to be thrown if no cart is found because, well you have an ID so you're supposed to get a cart. If you look for one, though ...